doc/source/reading.rst

```diff
@@ -0,0 +1,476 @@
+             :language: c++
+             :linenos:
+
+.. toggle-header::
+    :header: **Example Code Listing #3**
+
+    .. content-tabs::
+
+       .. tab-container:: cpp
+          :title: C++
+
+          .. literalinclude:: {source_examples_path}/cpp_api/reading_incomplete.cc
+             :language: c++
+             :linenos:
+
+
+Basic concepts and definitions
+------------------------------
+
+.. toggle-header::
+    :header: **Read query layout**
+
+      You can read from a TileDB array with different layouts (namely row-major,
+      column-major and global order). The row-/column-major layouts are
+      with respect to your defined subarray and may differ from the physical
+      layout of the cells on disk, attributing a lot of flexibility.
+
+.. toggle-header::
+    :header: **Maximum buffer sizes**
+
+      One great challenge when reading sparse arrays or arrays with
+      variable-length attributes (dense or sparse) is allocating the result
+      buffers to be set to the query. The reason is that
+      it is not possible to know the exact result size and, hence,
+      it is very hard to calculate how much memory you need to allocate
+      to your buffers. TileDB provides a function for computing a
+      good upper bound estimate on the buffer size that can certainly
+      hold the result for a given attribute.
+
+
+.. toggle-header::
+    :header: **Non-empty domain**
+
+      TileDB provides a useful function for determining the non-empty
+      domain of your array, i.e., a tight hyper-rectangle that encompasses
+      the non-empty cells stored in the (dense or sparse) array.
+
+.. toggle-header::
+    :header: **Incomplete queries**
+
+      In certain cases, you may wish to read a huge slice *incrementally*.
+      In such scenarios, allocating a huge buffer for the entire result
+      may not be feasible, whereas you may only have a fixed budget for
+      your buffers. TileDB enables you to submit queries on huge slices
+      with a buffer size budget, and notify you about whether your query
+      is completed or not. If it is incomplete, TileDB grants you a
+      partial result, which you can consume and then resume the query,
+      with TileDB incrementally bringing you the next partial results,
+      and so on.
+
+
+Reading in different layouts
+----------------------------
+
+TileDB allows you to retrieve the results of a subarray/slicing query in various
+layouts *with repsect to your specified subarray*, namely row-major, column-major
+and global order. You can set the layout to the query object as follows.
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      .. code-block:: c++
+
+        query.set_layout(TILEDB_ROW_MAJOR); // Can also be TILEDB_COL_MAJOR or TILEDB_GLOBAL_ORDER
+
+We demonstrate an example of a dense array using code listing #1 provided at
+the beginning of the tutorial. The figure below depicts the array contents and different
```

Comment:
"The figure below depicts the array contents and different result layouts." -> "The figure below depicts the array contents and the subarray read results for different query layouts. Notice that despite the global ordering of cells in the array, the read results are ordered with respect to the subarray of the query." I think the added second sentence is important, even though you stated it previously, to make sure the point is taken.